#164693398 Add the feature for users to be able to get all articles with pagination

codinger41 opened the pull request:
#### What does this PR do?
Adds the feature for users to be able to search and filter articles.
#### Description of Task to be completed?
Users should be able to get articles with pagination.
Users should be able to get  5 highest rated articles.
Users should be able to get 5 highest commented articles.
This new endpoint takes these queries:
- `limit` which is the number of articles to return, the default is 10.
- `offset` which is the number of articles to skip. the default is 0.

#### How should this be manually tested?
- Make a GET request to `/api/v1/articles` and add query parameters `limit` and `offset` with numbers as required.

- Make a GET request to  `/api/v1/articles/order` with the queries `type` which is the type of order, it can be `comments`, `ratings` or `latest`. 

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164693398

#### Screenshots (if appropriate)
#### Questions:
N/A

codinger41 committed:
feat(get_articles): implement endpoints to get articles with pagination support
- add a route to get all articles, with pagination support
- add a route to get articles by amount and criteriae

[Finishes #164693398]

codinger41 committed:
feat(get_articles): implement endpoints to get articles with pagination support
- add a route to get all articles, with pagination support
- add a route to get articles by amount and criteriae

[Finishes #164693398]

codinger41 committed:
Merge branch 'develop' into ft-get-articles-164693398

codinger41 committed:
feat(get_articles): implement endpoints to get articles

- add a route to get all articles, with pagination support
- add a route to get articles by amount and criteria

[Finishes #164693398]

Dubby20 commented:
Format your PR message

Dubby20 commented on controllers/articles.js:
```diff
@@ -267,4 +281,125 @@ export default class ArticleController {
+      }
+      return res.status(200).json({
+        success: true,
+        articles
```
No response message

codinger41 replied:
There's no need for a message if the request was successful @Dubby20

kleva-j replied:
If a request is successful you have to send a message to let the user know the status of their request.

9jaswag commented on controllers/articles.js:
```diff
@@ -267,4 +281,137 @@ export default class ArticleController {
+      } else {
+        articles = await Article.findAll({
+          where: {},
+          order: [[Comment, 'comment', 'ASC']],
```
@leksyib have you tested this to ensure it actually returns the result in the right order?

codinger41 replied:
Yes I have @9jaswag

9jaswag replied:
@leksyib  I'm surprised that `ASC` returns the right data. What does `DESC` return?

codinger41 replied:
`DESC` returns the data with the lowest below the highest. So the ones with the highest ratings come first and the lowest will come after

9jaswag commented on controllers/articles.js:
```diff
@@ -267,4 +281,112 @@ export default class ArticleController {
+  * @memberof ArticleController
+  * @returns {Object} order for findAll
+  */
+  static getOrder(type) {
```
this is slightly better. There are better alternatives but it sure beats the previous `if/else` logic.

9jaswag commented on controllers/articles.js:
```diff
@@ -273,4 +287,112 @@ export default class ArticleController {
+  * @memberof ArticleController
+  * @returns {Object} class instance
+  */
+  static async getArticlesByHighestField(req, res) {
```
@leksyib no change has been made here

9jaswag commented on controllers/articles.js:
```diff
@@ -273,4 +287,121 @@ export default class ArticleController {
+      `,
+      comments:
+      `
+        SELECT "Articles".*, COUNT("Comments".comment) AS avg_count
```
@leksyib this should be `comment_count`. We're not running an average on the count.

codinger41 replied:
Thanks I'll change that @9jaswag

9jaswag replied:
@leksyib this hasn't been changed yet

codinger41 replied:
@9jaswag  changed now

codinger41 committed:
feat(get_articles): implement endpoints to get articles  …
- add a route to get all articles, with pagination support
- add a route to get articles by amount and criteria

[Finishes #164693398]

codinger41 committed:
feat(get_articles): implement endpoints to get articles  …
- add a route to get all articles, with pagination support
- add a route to get articles by amount and criteria

[Finishes #164693398]
